src/client/dfs/common.c

```diff
@@ -635,8 +635,24 @@ entry_stat(dfs_t *dfs, daos_handle_t th, daos_handle_t oh, const char *name, siz
 int
 create_dir(dfs_t *dfs, dfs_obj_t *parent, daos_oclass_id_t cid, dfs_obj_t *dir)
 {
+	bool             exists;
+	struct dfs_entry entry = {0};
+	size_t           len   = strnlen(dir->name, DFS_MAX_NAME);
 	int rc;
 
+	if (parent != NULL) {
+		rc = fetch_entry(dfs->layout_v, parent->oh, dfs->th, dir->name, len, false, &exists,
+				 &entry, 0, NULL, NULL, NULL);
+		if (rc != 0) {
+			DL_ERROR(rc, "fetch_entry() %s failed.", dir->name);
+			return rc;
+		} else if (exists) {
+			if (S_ISDIR(entry.mode))
+				return EEXIST;
+			return ENOTDIR;
+		}
+	}
```

Comment:
the whole point of conditional inserts is to avoid doing an existence check to improve dfs IOPS performance. so this is not going to be acceptable.
the original comment above create_dir i believe was just left over before we had implemented conditional operations at the object level.

Reply:
I'll re-remove the check and also update the comment to avoid future confusion. :)